src/test/lib/optimizer/optimizer_test.cpp

```diff
@@ -331,4 +337,43 @@ TEST_F(OptimizerTest, OptimizesSubqueriesExactlyOnce) {
   }
 }
 
+TEST_F(OptimizerTest, CheckTrueCacheablility) {
+  auto optimizer = Optimizer::create_default_optimizer();
+
+  auto lqp = ProjectionNode::make(expression_vector(add_(b, subquery_a)),
+                                  PredicateNode::make(greater_than_(a, subquery_b), node_a));
+  const auto optimize_results = optimizer->optimize(std::move(lqp));
+  EXPECT_EQ(optimize_results.cacheable, true);
+}
+
+TEST_F(OptimizerTest, CheckFalseCacheabilityOfOptimization) {
+  auto optimizer = Optimizer{};
+  optimizer.add_rule(std::make_unique<JoinToSemiJoinRule>());
+
+  auto column_definitions = TableColumnDefinitions{};
+  column_definitions.emplace_back("column0", DataType::Int, false);
+  const auto table = std::make_shared<Table>(column_definitions, TableType::Data, ChunkOffset{2}, UseMvcc::Yes);
+
+  auto& sm = Hyrise::get().storage_manager;
+  sm.add_table("table", table);
```

Comment:
I'll pay for the increased storage cost if you rename the variable. Alternatively, as you don't use the variable a second time, just make it a oneliner.